### ISSUE TITLE
plusWildcard for mktap mail

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7602]
|<img alt="alexjacobson's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| alexjacobson reported|
|-|-|
|Trac ID|trac#366|
|Type|enhancement|
|Created|2003-10-24 02:15:23Z|

<details><summary>Searchable metadata</summary>

```
trac-id__366 366
type__enhancement enhancement
reporter__alexjacobson alexjacobson
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__ 
time__1066961723000000 1066961723000000
changetime__1469144720057342 1469144720057342
version__ 
owner__ 
cc__exarkun cc__alexjacobson
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="alexjacobson's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|alexjacobson commented|
|-|-|

```
#!html
<pre>
I wanted plus wildcarding in mktap.  So I added a few lines of code so that mail
to foo@host==foo+bar@host.  Not sure how to submit patches, but here are the
changes:
In twisted/mail/tap.py
* on line 55, I inserted (at optFlags):
          ["plusWildcard","+","foo+bar@host==foo@host"]
* on line 200 (or 201 after the above), I inserted:
        if config["plusWildcard"]: f.service.plusWildcard=1
In twisted/mail/protocols.py
* on line 69, I inserted:
        if hasattr(self.service,"plusWildcard") and self.service.plusWildcard:
            user.dest.local=user.dest.local.split("+")[0]

I hope that makes sense.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

```
#!html
<pre>
Without some more discriminating logic regarding when this rule is applied, this
feature won't work.  It will change the existing behavior in a scary way that
will break any email server _not_ expecting "plus wildcard" support.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|GitHub commented|
|-|-|

In changeset ad6a86f027e9c5a6d2edcd279626de83f6fe88a0
```
#!CommitTicketReference repository="" revision="ad6a86f027e9c5a6d2edcd279626de83f6fe88a0"
Merge pull request #366 from twisted/8636-rodrigc-itransport-doc

Author: rodrigc
Reviewer: scop
Fixes: #8636

Clarify docstring for ITransport.write() and ITransport.writeSequence()
```
